pkg/operator/staticpod/resourcecopy/source.go

```diff
@@ -0,0 +1,54 @@
+package resourcecopy
+
+// Source defines a source resource to be copied. The resource must have name, namespace and indicate whether it is optional or not.
+type Source interface {
+	Name() string
+	Namespace() string
+	IsOptional() bool
+}
```

Comment:
why does it need a source abstraction? It's equivalent to a struct with 3 fields.